spec/index.md

```diff
@@ -3781,15 +3859,19 @@ M.arg = candid(A)
 M.caller ∈ S.controllers[A.canister_id]
 
 if New_compute_allocation > S.compute_allocation[A.canister_id] or New_memory_allocation > S.memory_allocation[A.canister_id]:
-  freezing_limit(
-    New_compute_allocation,
-    New_memory_allocation,
-    New_freezing_threshold,
-    memory_usage_wasm_state(S.canisters[A.canister_id].wasm_state) +
-      memory_usage_raw_module(S.canisters[A.canister_id].raw_module) +
-      memory_usage_canister_history(New_canister_history),
-    S.canister_subnet[A.canister_id].subnet_size,
-  ) ≤ S.balances[A.canister_id]
+  liquid_balance(
+    S.balances[A.canister_id],
+    S.reserved_balances[A.canister_id],
+    freezing_limit(
+      New_compute_allocation,
+      New_memory_allocation,
```

Comment:
I guess a new memory allocation could also cause new cycles to become reserved here.